### PR TITLE
Feature 35 add embeddings addapter

### PR DESCRIPTION
**Pull request by gyfantidis**

### CODE REVIEW

**Review by sekasx — CHANGES_REQUESTED** [review]

File: gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/controller/DocumentController.java [hunk 1]
```diff
@@ -1,59 +1,176 @@
...
+                } catch (IOException | GendoxException e) {
+                    // Handle the exception (e.g., log the error or show an error message)
+                    e.printStackTrace();
+                    response.put("error", "Failed to upload or read the file: " + e.getMessage());
+                    return ResponseEntity.status(500).body(response); // 500 Internal Server Error
+                }
+            } else {
+                // Handle the case where the file is empty
+                response.put("error", "One or more files are empty");
+                return ResponseEntity.badRequest().body(response); // 400 Bad Request
```
[the remark this answers is not in the document]
Comment (gyfantidis) [reply gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/controller/DocumentController.java:166]: I deleted the else , now if there is an empty file it just skips it

[the remark this answers is not in the document]
Comment (gyfantidis) [reply gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/controller/DocumentController.java:166]: //            else {
//                // Handle the case where the file is empty
//                response.put("error", "One or more files are empty");
//                return ResponseEntity.badRequest().body(response); // 400 Bad Request
//            }
Deleted

File: gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/converters/EmbeddingGroupConverter.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package dev.ctrlspace.gendox.gendoxcoreapi.converters;
+
+import dev.ctrlspace.gendox.gendoxcoreapi.model.EmbeddingGroup;
+import dev.ctrlspace.gendox.gendoxcoreapi.model.dtos.EmbeddingGroupDTO;
+import org.springframework.stereotype.Component;
+
+@Component
+public class EmbeddingGroupConverter implements GendoxConverter<EmbeddingGroup, EmbeddingGroupDTO> {
```
[the remark this answers is not in the document]
Comment (gyfantidis) [reply gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/converters/EmbeddingGroupConverter.java:8]: We use EmbeddingGroupDTO

File: gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/controller/EmbeddingsController.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        return embeddingService.createSectionsEmbedding(sectionId);
+    }
+
+    @PostMapping("/embeddings/projects/{projectId}")
+    public List<Embedding> getProjectEmbeddings(@PathVariable UUID projectId) throws GendoxException {
+        return embeddingService.createProjectEmbeddings(projectId);
+    }
+
+    @PostMapping("/messages/semantic-search")
+    public List<DocumentInstanceSection> findCloserSections(@RequestBody MessageDto messageDto,
```
[the remark this answers is not in the document]
Comment (gyfantidis) [reply gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/controller/EmbeddingsController.java:73]: Ok I made it without using messageDTO, I haven't delete messageDTO.jave and messageConverter.java, maybe we need it in the future

File: gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/model/dtos/EmbeddingGroupDTO.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.UUID;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder(toBuilder = true)
+public class EmbeddingGroupDTO {
```
[the remark this answers is not in the document]
Comment (gyfantidis) [reply gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/model/dtos/EmbeddingGroupDTO.java:14]: Ok I made it without using `embeddingGroupDTO`, I haven't delete `embeddingGroupDTO.jave` and `embeddingGroupConverter.java`, maybe we need it in the future

File: gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/services/DocumentService.java [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+    public Page<DocumentInstance> getAllDocuments(DocumentCriteria criteria, Pageable pageable) throws GendoxException {
+        if (pageable == null) {
+            throw new GendoxException("Pageable cannot be null", "pageable.null", HttpStatus.BAD_REQUEST);
+        }
+        return documentInstanceRepository.findAll(DocumentPredicates.build(criteria), pageable);
+
+    }
+
+
+    public List<DocumentInstanceSection> getProjectSections(UUID projectId) throws GendoxException {
```
[the remark this answers is not in the document]
Comment (gyfantidis) [reply gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/services/DocumentService.java:81]: Create ticket #48

File: gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/controller/DocumentController.java [hunk 6]
```diff
@@ -1,59 +1,176 @@
...
-        // max total upload ~10MB configurable:
-        //spring.servlet.multipart.enabled=true
-        //spring.servlet.multipart.max-file-size=10MB
-        //spring.servlet.multipart.max-request-size=100MB
-        // Document section should be created for each file,
-        // different strategies should be supported, configurable from the DB
-        // https://www.digitalocean.com/community/tutorials/strategy-design-pattern-in-java-example-tutorial
-
-        throw new UnsupportedOperationException("Not implemented yet");
+    public ResponseEntity<Map<String, Object>> handleFilesUpload(@RequestParam("file") List<MultipartFile> files,
```
[the remark this answers is not in the document]
Comment (gyfantidis) [reply gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/controller/DocumentController.java:138]: create ticket #49

File: gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/controller/DocumentController.java [hunk 7]
```diff
@@ -1,59 +1,176 @@
...
+                                                                 @RequestParam("projectId") UUID projectId) {
+        Map<String, Object> response = new HashMap<>();
+
+
+        if (files.isEmpty()) {
+            // Handle the case where no files are provided
+            response.put("error", "No files provided");
+            return ResponseEntity.badRequest().body(response); // 400 Bad Request
+        }
+
```
[the remark this answers is not in the document]
Comment (gyfantidis) [reply gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/controller/DocumentController.java:149]: ok but when I upload a PDF it return as a Value this 
image

File: gendox-core-api/src/main/resources/application-local.yml [hunk 8]
```diff
@@ -23,6 +23,11 @@ spring:
   jackson:
     serialization:
       write-dates-as-timestamps: false
+  servlet:
+    multipart:
+      enabled: true
+      max-file-size: 10MB
+      max-request-size: 100MB
```
[the remark this answers is not in the document]
Comment (gyfantidis) [reply gendox-core-api/src/main/resources/application-local.yml:30]: I tasted, It works

**Pull request merged**